components/sup/src/manager/peer_watcher.rs

```diff
@@ -189,4 +190,36 @@ mod tests {
         }
         assert_eq!(expected_members, members);
     }
+
+    #[test]
+    fn with_file_using_poll_watcher() {
+        env::set_var("HAB_STUDIO_HOST_ARCH", "aarch64-darwin");
+        let tmpdir = TempDir::new().unwrap();
+        let path = tmpdir.path().join("some_other_file");
+        let mut file = OpenOptions::new().append(true)
+                                         .create_new(true)
+                                         .open(path.clone())
+                                         .unwrap();
+        let watcher = PeerWatcher::run(path).unwrap();
+        writeln!(file, "1.2.3.5:5").unwrap();
+        writeln!(file, "5.4.3.2").unwrap();
+        let member1 = Member { id: String::new(),
+                               address: String::from("1.2.3.5"),
+                               swim_port: 5,
+                               gossip_port: 5,
+                               ..Default::default() };
+        let member2 = Member { id: String::new(),
+                               address: String::from("5.4.3.2"),
+                               swim_port: GossipListenAddr::DEFAULT_PORT,
+                               gossip_port: GossipListenAddr::DEFAULT_PORT,
+                               ..Default::default() };
+
+        let expected_members = vec![member1, member2];
+        let mut members = watcher.get_members().unwrap();
+        for mut member in &mut members {
+            member.id = String::new();
+        }
+        assert_eq!(expected_members, members);
+        env::remove_var("HAB_STUDIO_HOST_ARCH");
+    }
```

Comment:
It seems that these tests need to be run in order otherwise the environmental variable will be incorrect.
By default rust will run cargo tests in multiple threads.
If I use the --test-threads=1 switch then the results are as expected.

Reply:
No.  The original was refactored by introducing the function peer_watcher_member_load_test().  These two tests use different values in for the Member constructors.  We could pass in the raw data into the above function as well, but I did not think that it added value or readability.